docs/how_backtrader_executes_code.rst

```diff
@@ -0,0 +1,105 @@
+TLDR: How BackTrader Executes Code
+==================================
+
+Preonce, Once, Next, Init – Why Isn’t My Code Working?
+------------------------------------------------------
+
+Backtrader evolved from a backtesting only platform to a platform that performs real-time
+processing of data. It now has signal generation from strategies, optimization and trade
+execution. Because of this there are different ways for Backtrader to execute indicator code
+varying depending on whether or not the data is live or historical and preloaded.
+
+Coding for Backtesting:
+
+This is how Backtrader processes indicator code when backtesting preloaded data.
+
+By LINES:
+
+	Init() - processes bars as lines, all at once
+
+	Nothing else needed
+
+
+By BARS:
+
+	Init() - processes bars as lines, all at once
+
+	Next() - processes different individual bars based on above bars
+
+Or
+
+By BARS:
+
+	Init() - empty or just cursory setup for next
+
+	Next() = Run for each bar to caculate
+
+
+By BATCH:
+
+	Once - processes an array that you step through for	speed
+
+	It ends. It doesn’t run next(), doesn’t run init(), nothing. This is for speed
+
```

Comment:
Probably need to explain in what scenario each mode is used ( what the Backtrader user needs to do to trigger each mode)

Reply:
Good point. Could be another chart showing "how it's run" and then what path it takes.